lib/urlutil.js

```diff
@@ -25,37 +25,50 @@ exports.expand = (tpl, data) => {
   let params = {};
   Object.keys(TPL_PARAMS).forEach(key => {
     let val = data[TPL_PARAMS[key]];
-    if (TPL_TRANSFORMS[key]) {
-      val = TPL_TRANSFORMS[key](val, params);
-    }
     if (val !== null && val !== undefined) {
       params[key] = val;
     }
   });
+
+  Object.keys(TPL_TRANSFORMS).forEach(key => {
```

Comment:
I'm doing this in a second pass after so all the params are available to pass in to the transform, not just the ones before this one in the list.

Reply:
Makes sense.  Good sense.